lib/dynamo/dynamo/jobs.py

```diff
@@ -232,6 +235,36 @@ def update_job(job: dict) -> None:
     )
 
 
+# TODO:
+#  - add dynamo and api tests for updating name when the job doesn't have one
+#  - allow updating arbitrary fields? and handle name=None as special case?
+def update_job_for_user(job_id: str, name: str | None, user_id: str) -> dict:
+    """Update the user's job at their request."""
+    if name is not None:
+        update_expression = 'SET #name = :name'
+        name_value = {':name': name}
+    else:
+        update_expression = 'REMOVE #name'
+        name_value = {}
+
+    table = DYNAMODB_RESOURCE.Table(environ['JOBS_TABLE_NAME'])
+    try:
+        job = table.update_item(
+            Key={'job_id': job_id},
+            UpdateExpression=update_expression,
+            ConditionExpression='user_id = :user_id',
+            ExpressionAttributeValues={':user_id': user_id, **name_value},
+            ExpressionAttributeNames={'#name': 'name'},
+            ReturnValues='ALL_NEW',
+        )['Attributes']
+    except botocore.exceptions.ClientError as e:
+        if e.response['Error']['Code'] == 'ConditionalCheckFailedException':
+            raise UpdateJobForDifferentUserError("You cannot modify a different user's job")
+        raise
```

Comment:
```suggestion
        raise e
```

Reply:
Bare `raise` already re-raises the current exception, so the same as `raise e` in this case. I'd prefer to keep it as-is for consistency with the functions in [dynamo.user](https://github.com/ASFHyP3/hyp3/blob/develop/lib/dynamo/dynamo/user.py), which also follow this pattern.

Though, I should probably add a unit test for the `raise` case. I recall running into difficulty when trying to add a test for this last week, but I don't remember what the issue was, so I'll try again.

Reply:
Added a unit test for the `raise` statement.

Reply:
Cool, didn't know that. I though you needed to pass the exception.